source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitor.cs

```diff
@@ -103,22 +117,22 @@ internal async Task OnNewEvent(WatchEventType type, V1Pod pod)
                 {
                     case WatchEventType.Added or WatchEventType.Modified:
                     {
-                        if (!podStatusLookup.TryGetValue(scriptTicket, out var status))
-                        {
-                            status = new PodStatus(pod.GetScriptTicket());
-                            podStatusLookup[scriptTicket] = status;
-                        }
-
+                        var status = podStatusLookup.GetOrAdd(scriptTicket, st => new PodStatus(st, clock));
                         status.Update(pod);
                         log.Verbose($"Updated pod {pod.Name()} status. {status}");
 
                         break;
                     }
                     case WatchEventType.Deleted:
-                        log.Verbose($"Removed {type} pod {pod.Name()} status");
-
                         //if the pod is deleted, remove it
-                        podStatusLookup.Remove(scriptTicket);
+                        if (podStatusLookup.TryRemove(scriptTicket, out _))
```

Comment:
handling a failure for Try remove by logging a warning. Any missed pods will be cleaned up at the refresh (every 10 mins).